lib/collection/src/operations/types.rs

```diff
@@ -328,6 +328,27 @@ pub struct ShardTransferInfo {
     #[serde(skip_serializing_if = "Option::is_none")]
     #[anonymize(false)]
     pub comment: Option<String>,
+
+    /// Internal status of the shard transfer, do not expose.
+    #[serde(skip)]
+    #[schemars(skip)]
+    #[anonymize(false)]
+    pub status: ShardTransferStatus,
+}
```

Comment:
Nits:
- reword the doc to make it clear that `comment` and `status` contain the same information.
- `#[serde(skip)]` implies `#[schemars(skip)]`, so the latter could be removed.
- Put into option (see comment below).

```suggestion
    /// Same as [`Self::comment`], but in a machine-readable form.
    /// For internal use only, do not expose.
    #[serde(skip)]
    #[anonymize(false)]
    pub status: Option<ShardTransferStatus>,
}
```